angular/headless/src/lib/coercion.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Transforms a value (typically a string) to a boolean.
+ * Intended to be used as a transform function of an input.
+ *
+ *  @usageNotes
+ *  ```typescript
+ *  @Input({ transform: auBooleanAttribute }) status: boolean | undefined;
+ *  ```
+ * @param value Value to be transformed.
+ */
+export function auBooleanAttribute(value: unknown): boolean | undefined {
+	if (value === undefined) {
+		return undefined;
+	}
+	return typeof value === 'boolean' ? value : value != null && value !== 'false';
+}
+
+/**
+ * Transforms a value (typically a string) to a number.
+ * Intended to be used as a transform function of an input.
+ * @param value Value to be transformed.
+ *
+ *  @usageNotes
+ *  ```typescript
+ *  @Input({ transform: numberAttribute }) id: number | undefined;
+ *  ```
+ */
+export function auNumberAttribute(value: unknown): number | undefined {
+	// parseFloat(value) handles most of the cases we're interested in (it treats null, empty string,
+	// and other non-number values as NaN, where Number just uses 0) but it considers the string
+	// '123hello' to be a valid number. Therefore we also check if Number(value) is NaN.
+	if (value === undefined) {
+		return undefined;
+	}
+	const isNumberValue = !isNaN(parseFloat(value as any)) && !isNaN(Number(value));
+	return isNumberValue ? Number(value) : NaN;
```

Comment:
Is there any reason why we do not do simply `return +value`?

Reply:
Maybe because of null and empty string...

Reply:
So maybe we could simply do:
```ts
return (value === null || value === "") ? NaN : +value;
```

Reply:
I have to confess I put my brain on the side and copied [angular's implementation](https://github.com/angular/angular/blob/main/packages/core/src/util/coercion.ts)...

I also hesitated to write instead:
```typescript
export function auNumberAttribute(value: unknown): number | undefined {
	if (value === undefined) {
		return undefined;
	}
	return numberAttribute(value);
}
```
What do you think ?

Reply:
I prefer that indeed.

Reply:
ok for me !